internal/services/computefleet/compute_fleet_resource.go

```diff
@@ -0,0 +1,1761 @@
+			Optional:     true,
+			ValidateFunc: validation.FloatAtLeast(0.0),
+			AtLeastOneOf: []string{"vm_attributes.0." + parent + ".0.max", "vm_attributes.0." + parent + ".0.min"},
+		},
+
+		"min": {
+			Type:         pluginsdk.TypeFloat,
+			Optional:     true,
+			ValidateFunc: validation.FloatAtLeast(0.0),
+			AtLeastOneOf: []string{"vm_attributes.0." + parent + ".0.max", "vm_attributes.0." + parent + ".0.min"},
+		},
+	}
+}
+
+func (r ComputeFleetResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"unique_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r ComputeFleetResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ComputeFleet.ComputeFleetClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ComputeFleetResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := fleets.NewFleetID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := fleets.Fleet{
+				Location: location.Normalize(model.Location),
+				Plan:     expandPlanModel(model.Plan),
+				Properties: &fleets.FleetProperties{
+					RegularPriorityProfile: expandRegularPriorityProfileModel(model.RegularPriorityProfile),
+					SpotPriorityProfile:    expandSpotPriorityProfileModel(model.SpotPriorityProfile),
+					VMAttributes:           expandVMAttributesModel(model.VMAttributes),
+					VMSizesProfile:         pointer.From(expandVMSizeProfileModel(model.VMSizesProfile)),
+				},
+			}
+
+			if model.Tags != nil {
+				properties.Tags = pointer.To(model.Tags)
+			}
+
+			if model.Zones != nil {
+				properties.Zones = pointer.To(model.Zones)
+			}
+
+			expandedIdentity, err := identity.ExpandLegacySystemAndUserAssignedMapFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+			properties.Identity = expandedIdentity
+
+			additionalLocationsProfileValue, err := expandAdditionalLocationProfileModel(model.AdditionalLocationProfile, metadata.ResourceData)
+			if err != nil {
+				return err
+			}
+			properties.Properties.AdditionalLocationsProfile = additionalLocationsProfileValue
+
+			computeProfile := fleets.ComputeProfile{
+				AdditionalVirtualMachineCapabilities: &fleets.AdditionalCapabilities{
+					HibernationEnabled: pointer.To(model.AdditionalCapabilitiesHibernationEnabled),
+					UltraSSDEnabled:    pointer.To(model.AdditionalCapabilitiesUltraSSDEnabled),
+				},
```

Comment:
could we create a block in schema to match the API as well improve ux?

Reply:
Yes, fixed.